lms/djangoapps/completion/models.py

```diff
@@ -106,6 +106,86 @@ def submit_completion(self, user, course_key, block_key, completion):
         return obj, isnew
 
 
+class AggregateCompletionManager(models.Manager):
+    """
+    Custom manager for AggregateCompletion model.
+
+    """
+
+    def submit_completion(self, user, course_key, block_key, aggregation_name, earned, possible):
+        """
+        Inserts and Updates the completion AggregateCompletion for the specified record.
+
+        Parameters:
+            * user (django.contrib.auth.models.User): The user for whom the
+              completion is being submitted.
+            * course_key (opaque_keys.edx.keys.CourseKey): The course in
+              which the submitted block is found.
+            * block_key (opaque_keys.edx.keys.UsageKey): The block that has had
+              its completion changed.
+            * aggregation_name : The name of the aggregated blocks. This is set by the level that the aggregation
+              is occuring. Possible values are "course", "chapter", "sequential", "vertical"
+            * earned: The sum of the fractional completions of all descendant blocks
+            * possible: The total sum of the possible completion values of all descendant
+              blocks that are visible to the user. This will always be a positive integer.
+
+        Return Value:
+            (BlockCompletion, bool): A tuple comprising the created or updated
+            BlockCompletion object and a boolean value indicating whether the
+            object was newly created by this call.
+
+        Raises:
+
+            ValueError:
+                If the wrong type is passed for one of the parameters.
+
+            django.core.exceptions.ValidationError:
+                If a float is passed that is not between 0.0 and 1.0.
+
+            django.db.DatabaseError:
+                If there was a problem getting, creating, or updating the
+                BlockCompletion record in the database.
+
+                This will also be a more specific error, as described here:
+                https://docs.djangoproject.com/en/1.11/ref/exceptions/#database-exceptions.
+                IntegrityError and OperationalError are relatively common
+                subclasses.
+        """
+
+        # Raise ValueError to match normal django semantics for wrong type of field.
+        if not isinstance(course_key, CourseKey):
```

Comment:
Please check that the `block_key` is a `UsageKey`, like you're doing here with `CourseKey`.